app/assets/javascripts/upload/multiFileUpload.js

```diff
@@ -609,7 +610,13 @@ export class MultiFileUpload {
             method: 'POST',
             body: formData
         })
-            .then(response => response.json())
+            .then(response => {
+                              if (response.status == 500) {
+                              window.location.href = this.getErrorServiceUrl();
+                              } else {
+                                return response.json();
+                              }
```

Comment:
It'd be nice to have some tests for this scenario in the JS spec

Reply:
Unfortunately, the attempts made to write a test scenario for this din't work. Looks like the 'fetch' method is not properly called/mocked. 
One of such unsuccessful attempts: 
` describe("When upscan error response is 500", () => {
            const errorResponse = new Response(JSON.stringify({}), { status: 500, statusText: 'Internal Server Error', });
                beforeEach((done) => {
                            instance = new MultiFileUpload(container);
                            spyOn(instance, 'setDuplicateInsetText').and.returnValue(Promise.resolve({}));
                            spyOn(instance, 'requestProvisionUpload').and.callThrough((file));
                            spyOn(instance, 'fetch').and.resolveTo(errorResponse);
                            instance.init();
                        });
                it('statusCode=500', (done) => {
                expect(instance.requestProvisionUpload).toHaveBeenCalled();
                done();
                });
            });` 

Please share if there's any reference for such test case.

Reply:
Thanks 😃 , added the test.